components/kserve/kserve_config_handler.go

```diff
@@ -0,0 +1,163 @@
+		}
+		if k.DefaultDeploymentMode == "" {
+			fmt.Println("Serving is removed, Kserve will default to rawdeployment")
+		}
+		if err := k.setDefaultDeploymentMode(ctx, cli, dscispec, RawDeployment); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (k *Kserve) setDefaultDeploymentMode(ctx context.Context, cli client.Client, dscispec *dsciv1.DSCInitializationSpec, defaultmode DefaultDeploymentMode) error {
+	inferenceServiceConfigMap := &corev1.ConfigMap{}
+	err := cli.Get(ctx, client.ObjectKey{
+		Namespace: dscispec.ApplicationsNamespace,
+		Name:      KserveConfigMapName,
+	}, inferenceServiceConfigMap)
+	if err != nil {
+		return fmt.Errorf("error getting configmap 'inferenceservice-config'. %w", err)
+	}
+
+	// set data.deploy.defaultDeploymentMode to the model specified in the Kserve spec
+	var deployData map[string]interface{}
+	if err = json.Unmarshal([]byte(inferenceServiceConfigMap.Data["deploy"]), &deployData); err != nil {
+		return fmt.Errorf("error retrieving value for key 'deploy' from configmap %s. %w", KserveConfigMapName, err)
+	}
+	deployData["defaultDeploymentMode"] = defaultmode
+	deployDataBytes, err := json.MarshalIndent(deployData, "", " ")
+	if err != nil {
+		return fmt.Errorf("could not set values in configmap %s. %w", KserveConfigMapName, err)
+	}
+	inferenceServiceConfigMap.Data["deploy"] = string(deployDataBytes)
+
+	var ingressData map[string]interface{}
+	if err = json.Unmarshal([]byte(inferenceServiceConfigMap.Data["ingress"]), &ingressData); err != nil {
+		return fmt.Errorf("error retrieving value for key 'ingress' from configmap %s. %w", KserveConfigMapName, err)
+	}
+	if defaultmode == RawDeployment {
+		ingressData["disableIngressCreation"] = true
+	} else {
+		ingressData["disableIngressCreation"] = false
+	}
+	ingressDataBytes, err := json.MarshalIndent(ingressData, "", " ")
+	if err != nil {
+		return fmt.Errorf("could not set values in configmap %s. %w", KserveConfigMapName, err)
+	}
+	inferenceServiceConfigMap.Data["ingress"] = string(ingressDataBytes)
+
+	if err = cli.Update(ctx, inferenceServiceConfigMap); err != nil {
+		return fmt.Errorf("could not set default deployment mode for Kserve. %w", err)
+	}
+
+	return nil
+}
+
+func (k *Kserve) configureServerless(instance *dsciv1.DSCInitializationSpec) error {
+	switch k.Serving.ManagementState {
+	case operatorv1.Unmanaged: // Bring your own CR
+		fmt.Println("Serverless CR is not configured by the operator, we won't do anything")
+
+	case operatorv1.Removed: // we remove serving CR
+		fmt.Println("existing Serverless CR (owned by operator) will be removed")
+		if err := k.removeServerlessFeatures(instance); err != nil {
+			return err
+		}
+
+	case operatorv1.Managed: // standard workflow to create CR
+		switch instance.ServiceMesh.ManagementState {
+		case operatorv1.Unmanaged, operatorv1.Removed:
+			return fmt.Errorf("ServiceMesh is need to set to 'Managed' in DSCI CR, it is required by KServe serving field")
+		}
+
```

Comment:
A call to `checkDependentOperators` is missing here. Looks like a rebase mistake.

Reply:
should be fixed now